lib/active_model_serializers/adapter/json_api/pagination_links.rb

```diff
@@ -53,11 +53,16 @@ def last_page_url
 
         def prev_page_url
           return nil if collection.current_page == FIRST_PAGE
-          url_for_page(collection.current_page - FIRST_PAGE)
+
+          if collection.current_page > collection.total_pages
+            url_for_page(collection.total_pages)
+          else
+            url_for_page(collection.current_page - FIRST_PAGE)
+          end
         end
 
         def next_page_url
-          return nil if collection.total_pages == 0 || collection.current_page == collection.total_pages
+          return nil if collection.total_pages == 0 || collection.current_page >= collection.total_pages
           url_for_page(collection.next_page)
```

Comment:
So no `next page` if the `current page` is the last page or out of bounds.

Reply:
That's correct. This way if you request out of bounds and rely on the next link, you don't paginate forever.